CRM/UF/Page/ProfileEditor.php

```diff
@@ -192,7 +192,10 @@ public static function getSchema($entityTypes) {
           break;
 
         default:
-          throw new CRM_Core_Exception("Unrecognized entity type: $entityType");
+          CRM_Utils_Hook::profileSchemas($civiSchema, $entityType, $availableFields);
```

Comment:
So in your use-case case is  `$entityType == 'GrantModel'`? And could we just abstract this pattern a bit and make it trim the word "Model" off the string and do like the others do above?

Reply:
Pretty much

Reply:
Hmm - looks like the hook wouldn't be needed with a bit of code cleanup